### PR TITLE
Refactor FE Functional Tests for 'page contains following fields' Step

### PR DESCRIPTION
**Pull request by noronhaa**
Number of changes to the Functional test framework:

- Implement way of parsing data from a page object class back to a calling class so page data can be compared / used further down the test layers
- remove ID being parsed in as a parameter into basic ccd component dataType classes (we already pass css, no need to have separate param for id) . this param has been replaced by 'key' param to identify a field
- use this new functionality to implement step 'the page contains the following fields'
- abstract complex types into their own classes to improve reusability and fall more in line with the page object model 
- Removed all use of html ID's anywhere other than page objects. This includes removing ID's from Step classes and Scenario/Feature files them selves and has required refactoring of tests and classes

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-5208

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage1.js [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
+let CCDStringField = require('../../ccd-components/fields/ccdStringField.js');
 
 class ConditionalsCreateCasePage1WizardPage extends CreateCaseWizardPage {
 
   constructor() {
     super();
     this._fieldUsedInShowCondition = ''; // TextField | TextFieldOptional
-    this.mandatoryTextField = new CCDStringField('ccd-write-text-field', 'text', 'TextField');
-    this.optionalTextField = new CCDStringField('ccd-write-text-field', 'text', 'TextFieldOptional');
+    this.TextField = new CCDStringField('#TextField');
```
Comment (rafalkalita) [inline test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage1.js]: variable names should not start with a capital letter

File: test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage2.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+let CCDStringField = require('../../ccd-components/fields/ccdStringField.js');
+
+class ConditionalsCreateCasePage2WizardPage extends CreateCaseWizardPage {
+
+  constructor() {
+    super();
+    this.TextField3 = new CCDStringField('#TextField3');
```
Comment (rafalkalita) [inline test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage2.js]: textField3 - lower first letter please

File: test/utils/fieldUtils.js [hunk 3]
```diff
@@ -506,7 +392,7 @@ class FieldDataTypes {
   }
 
   /**
-   * Check if field is present and enabled
+   * Check if field is present and enabledstring
```
Comment (rafalkalita) [inline test/utils/fieldUtils.js]: Typo

File: test/features/complexTypes.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
+@functional @complexcollections @showandhide @rerun
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/features/complexTypes.feature]: ahh should of been deleted - will remove - was me re running failing tests locally

File: test/features/complexTypes.feature [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Feature: Set of scenarios testing show & hide functionality and conditional logic
+
+  Background:
+    Given I have logged in
+
+  Scenario: limits the complex type using the EventToComplexTypes sheet
+    Given a case type containing Complex and Collection types exists
+    When I populate the form with the school data
+#    Then only the fields defined in EventToComplexTypes sheet should be visible
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/features/complexTypes.feature]: done

File: test/pageObjects/ccd-components/complexTypes/addressComplex.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+class AddressComplex extends ComplexBase {
+
+  constructor() {
+    super(id);
+    this.id = id;
+    this.initialiseFields(this.id)
+  }
+
+  /**
+   * Allows us to also re-initialise fields fpr a collection by parsing collection parent id here
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/complexTypes/addressComplex.js]: done

File: test/pageObjects/ccd-components/complexTypes/addressComplex.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+   * @param id - id of the complex OR may be the id of the collection (of this complex)
+   * @param collectionOrderIndex
+   */
+  initialiseFields(id,collectionOrderIndex) {
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/complexTypes/addressComplex.js]: done

File: test/pageObjects/ccd-components/complexTypes/addressHiddenComplex.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+class AddressHiddenComplex extends ComplexBase {
+
+  constructor() {
+    super(id);
+    this.id = id;
+    this.initialiseFields(this.id)
+  }
+
+  /**
+   * Allows us to also re-initialise fields fpr a collection by parsing collection parent id here
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/complexTypes/addressHiddenComplex.js]: done

File: test/pageObjects/ccd-components/fields/ccdCollection.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+   * Click the 'Add New' button within this Collection
+   * @returns {Promise<void>}
+   */
+  async clickAddNewButton(){
+    await this.addNewButton.click();
+  }
+
+  /**
+   * Remove a specific item in the collection
+   * @param collectionItemInt - tem in collection (eg 1st field/complex) to remove
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/fields/ccdCollection.js]: done

File: test/pageObjects/ccd-components/fields/ccdDocumentField.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+/**
+ * CCD Fixed Radio List field component
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/fields/ccdDocumentField.js]: ahh good spot. done

File: test/pageObjects/ccd-components/fields/ccdDocumentField.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+  setKey(key){
+    if (typeof key === 'undefined') {
+      return this.css.replace('#','');
+    } else {
+      return key;
+    }
+  }
+  /**
+   * Returns an options array
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/fields/ccdDocumentField.js]: done

File: test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage1.js (hunk 1 above)
Comment (noronhaa) [reply test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage1.js]: done

File: test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage2.js (hunk 2 above)
Comment (noronhaa) [reply test/pageObjects/wizardPages/Conditionals/conditionals_CreateCase_ConditionalPage2.js]: done

File: test/utils/fieldUtils.js (hunk 3 above)
Comment (noronhaa) [reply test/utils/fieldUtils.js]: done

File: test/pageObjects/ccd-components/fields/ccdFixedList.js [hunk 12]
```diff
@@ -1,31 +1,54 @@
-
+let BasePage = require('../../basePage');
 let Dropdown = require('../../webdriver-components/dropdown.js');
 
 /**
  * CCD Fixed List dropdown field component
  */
-class CcdFixedList {
+class CcdFixedList extends Dropdown{
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/pageObjects/ccd-components/fields/ccdFixedList.js]: done

**Pull request merged**